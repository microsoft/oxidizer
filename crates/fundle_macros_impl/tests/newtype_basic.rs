// Copyright (c) Microsoft Corporation.

use syn::{parse_quote,  ItemStruct};

mod util;

#[test]
fn basic_expansion() {
    let item: ItemStruct = parse_quote! {
        #[newtype]
        struct Foo(String);
    };

    insta::assert_snapshot!(expand_fundle_newtype!(item));
}