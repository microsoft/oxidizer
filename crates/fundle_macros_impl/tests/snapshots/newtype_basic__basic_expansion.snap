---
source: crates/fundle_macros_impl/tests/newtype_basic.rs
assertion_line: 14
expression: expand_fundle_newtype!(item)
---
#[derive(Clone)]
#[allow(dead_code)]
struct Foo(String);
impl<T> From<T> for Foo
where
    T: AsRef<String>,
    String: Clone,
{
    fn from(x: T) -> Self {
        Self(x.as_ref().clone())
    }
}
impl std::ops::Deref for Foo {
    type Target = String;
    fn deref(&self) -> &Self::Target {
        &self.0
    }
}
impl std::ops::DerefMut for Foo {
    fn deref_mut(&mut self) -> &mut Self::Target {
        &mut self.0
    }
}
